### PR TITLE
Users crud V3

### PR DESCRIPTION
**Pull request by LazarShockX**
Updated admin page with the following features:

- Admin toggle button
- Application status
- Approve or Reject content creator application
- Delete user
- Adjust rows per page
- Admin page and route are only accessible to admins
- General UI overhaul to replicate the one in Figma

Below is a video showcasing all features except bullet point six.

https://github.com/user-attachments/assets/8b9b6e34-b5d9-4254-8f39-75cfb68876aa

All acceptance criteria have bee fulfilled 🩵

### CODE REVIEW

**Review by LazarShockX — COMMENTED** [review]
Merge successful 🩵

**Comment by LazarShockX** [comment]
Screenshot 2024-10-30 at 14 26 45
Screenshot 2024-10-30 at 14 30 06

The 2 failing Cypress Tests are incorrect, since the application overview page correctly shows a list of applicants and the search term works.

**Review by madsWesth — APPROVED** [review]

**Comment by LazarShockX** [comment]
En gruppe syntes det var sjovt at flytte Navbar til app/src/components/navbar/Navbar.tsx, så nu har vi en ny fucking merge konflikt

File: app/src/services/auth.guard.ts
Comment (Laskiri) [inline app/src/services/auth.guard.ts:1]: I would suggest, making a variable isAdmin = userRole === requiredRole, and use that in your logical expressions for better readability if it's not to much work ^^,

**Review by Laskiri — CHANGES_REQUESTED** [review]
Remove the redundant path, and AdminProtect the others as i've described on the app.tsx file. Additionally take a look at my other suggestions if you're not to stressed with other things, else its small fixes that can be handled in the future.

**Comment by Laskiri** [comment]
> En gruppe syntes det var sjovt at flytte Navbar til app/src/components/navbar/Navbar.tsx, så nu har vi en ny fucking merge konflikt og det værste er, at jeg ikke engang kan fikse det i web editoren

sig til hvis du får brug for hjælp til dette

**Comment by LazarShockX** [comment]
> > En gruppe syntes det var sjovt at flytte Navbar til app/src/components/navbar/Navbar.tsx, så nu har vi en ny fucking merge konflikt og det værste er, at jeg ikke engang kan fikse det i web editoren
> 
> sig til hvis du får brug for hjælp til dette

Meget gerne 🩵 for når jeg prøver at løse det, opstår der 42 andre konflikter
Screenshot 2024-11-01 at 10 36 26

**Review by Todua992 — APPROVED** [review]
Honestly it looks good, Nice touch with UI. And yeah, I agree with Lasse about the Request change he made. The rest of the code seems to work, and doesnt break application. So nice work with that.

**Comment by LazarShockX** [comment]
> Remove the redundant path, and AdminProtect the others as i've described on the app.tsx file. Additionally take a look at my other suggestions if you're not to stressed with other things, else its small fixes that can be handled in the future.

I will look into your suggestions for the next sprint. Right now I just want to merge it 🩵

**Comment by LazarShockX** [comment]
Det er helt væk. Den har ikke merget usersCrud ind i dev.

**Comment by Laskiri** [comment]
> > Remove the redundant path, and AdminProtect the others as i've described on the app.tsx file. Additionally take a look at my other suggestions if you're not to stressed with other things, else its small fixes that can be handled in the future.
> 
> I will look into your suggestions for the next sprint. Right now I just want to merge it 🩵

No worries, as i said most if it was minor things for better quality and readabily, however the AdminProtectedRoutes must be integrated properly with the different paths. Good job with this pbi guys, i know it was really large but you got around with it 💪

**Pull request merged**